acvm-repo/brillig/src/black_box.rs

```diff
@@ -61,45 +42,32 @@ pub enum BlackBoxOp {
         input2_infinite: MemoryAddress,
         result: HeapArray,
     },
-    BigIntAdd {
-        lhs: MemoryAddress,
-        rhs: MemoryAddress,
-        output: MemoryAddress,
-    },
-    BigIntSub {
-        lhs: MemoryAddress,
-        rhs: MemoryAddress,
-        output: MemoryAddress,
-    },
-    BigIntMul {
-        lhs: MemoryAddress,
-        rhs: MemoryAddress,
-        output: MemoryAddress,
-    },
-    BigIntDiv {
-        lhs: MemoryAddress,
-        rhs: MemoryAddress,
-        output: MemoryAddress,
-    },
-    BigIntFromLeBytes {
-        inputs: HeapVector,
-        modulus: HeapVector,
-        output: MemoryAddress,
-    },
-    BigIntToLeBytes {
-        input: MemoryAddress,
-        output: HeapVector,
-    },
-    Poseidon2Permutation {
-        message: HeapVector,
-        output: HeapArray,
-        len: MemoryAddress,
-    },
-    Sha256Compression {
-        input: HeapArray,
-        hash_values: HeapArray,
-        output: HeapArray,
-    },
+    /// BigInt addition
+    BigIntAdd { lhs: MemoryAddress, rhs: MemoryAddress, output: MemoryAddress },
+    /// BigInt subtraction
+    BigIntSub { lhs: MemoryAddress, rhs: MemoryAddress, output: MemoryAddress },
+    /// BigInt multiplication
+    BigIntMul { lhs: MemoryAddress, rhs: MemoryAddress, output: MemoryAddress },
+    /// BigInt division
+    BigIntDiv { lhs: MemoryAddress, rhs: MemoryAddress, output: MemoryAddress },
+    /// BigInt from le bytes
+    BigIntFromLeBytes { inputs: HeapVector, modulus: HeapVector, output: MemoryAddress },
+    /// BigInt to le bytes
+    BigIntToLeBytes { input: MemoryAddress, output: HeapVector },
+    /// Applies the Poseidon2 permutation function to the given state,
+    /// outputting the permuted state.
+    Poseidon2Permutation { message: HeapVector, output: HeapArray, len: MemoryAddress },
+    /// Applies the SHA-256 compression function to the input message
+    Sha256Compression { input: HeapArray, hash_values: HeapArray, output: HeapArray },
+    /// Returns a decomposition in `num_limbs` limbs of the given input over the given radix.
+    ///
+    /// - The value stored in `radix` must be in the range [2, 256]
+    /// - `num_limbs` must be at least one if the value stored in `input` is not zero.
+    /// - The value stored in `output_bits` must have a `bit_size` of one.
+    ///   That value specifies whether we should decompose into bits. The value stored in
+    ///   the `radix` address is expected to be two if the value stored in `output_bits` is equal to one.
```

Comment:
```suggestion
    ///   That value specifies whether we should decompose into bits. The value stored in
    ///   the `radix` address is must be two if the value stored in `output_bits` is equal to one.
```

In general we should avoid "is expected" language in cases where we require a certain property to hold.